### PR TITLE
Initial commit for edgexfoundry-holding repository.

### PR DESCRIPTION
**Pull request by kb2ma**
Signed-off-by: Ken Bannister <kb2ma@runbox.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-examples/blob/master/.github/CONTRIBUTING.md.

## What is the current behavior?

new repository

## Issue Number:
n/a

## What is the new behavior?
Initial commit taken from from [kb2ma/device-coap-c-pre-holding](https://github.com/kb2ma/device-coap-c-pre-holding). See README.md for details.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports
n/a

## Specific Instructions
See inline comments for potential issues

## Other information

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+clean:
+	    rm -f $(MICROSERVICES)
+
+docker: $(DOCKERS)
+
+docker_device_coap_c:
+	    docker build \
+	        -f scripts/Dockerfile.alpine-3.11 \
+	        --label "git_sha=$(GIT_SHA)" \
+	        -t kb2ma/docker-device-coap-c:${GIT_SHA} \
```
Comment (kb2ma) [inline Makefile:30]: Change the tag?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
+# device-coap-c
+
+EdgeX device service for CoAP-based REST protocol
+
+This device service allows a 3rd party sensor application to push data into EdgeX via CoAP. Like HTTP, CoAP provides REST based access to resources, but CoAP is more compact for use in constrained IoT devices.
+
+The device-coap-c service (_device-coap_ for short) is modeled after the HTTP based [device-rest-go](https://github.com/edgexfoundry/device-rest-go) service, and runs over UDP. The current implementation is meant for one-way communication from a device into EdgeX by posting readings asynchronously. Synchronous interaction initiated by EdgeX can be added in the future.
+
+device-coap uses DTLS for secure communication to devices. It is written in C, and relies on the well known [libcoap](https://libcoap.net/) library.
+
```
Comment (kb2ma) [inline README.md:10]: Add link to meeting and slides from EdgeX Device Working Group on 2020-10-12?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+Below is an example entry for a docker-compose template with the rest of the EdgeX setup. The CoAP server listens on the default secure port, 5684. It also listens on any interface since the CoAP message likely arrives from an external network.
+
+```
+  device-coap:
+    image: kb2ma/docker-device-coap-c:0.2-dev
+    ports:
+      - "127.0.0.1:49750:49750"
+      - "0.0.0.0:5684:5684/udp"
+    container_name: kb2ma-device-coap
```
Comment (kb2ma) [inline README.md:101]: Container name OK?

File: res/configuration-native.toml [hunk 4]
```diff
@@ -0,0 +1,66 @@
+# For running a native executable of device-coap, rather than in a container.
+# Changes Host entries
+[Service]
+  BootTimeout = 30000
+  CheckInterval = '10s'
+  Host = '172.17.0.1'
+  ServerBindAddr = ''
+  Port = 49750
```
Comment (kb2ma) [inline res/configuration-native.toml:8]: Need to assign a port?

File: VERSION [hunk 5]
```diff
@@ -0,0 +1 @@
+0.2-dev
```
Comment (kb2ma) [inline VERSION:1]: Once we work through issues with this PR, I plan to tag as 0.2. Let me know of any guidelines around version numbers.

**Review by kb2ma — COMMENTED** [review]
Added inline comments for open questions.

File: Makefile (hunk 1 above)
Comment (iain-anderson) [reply Makefile:30]: edgexfoundry/docker-device-coap-c:${GIT_SHA}

File: README.md (hunk 2 above)
Comment (iain-anderson) [reply README.md:10]: Sure. They can be found on https://wiki.edgexfoundry.org/display/FA/Device+and+Device+SDK+Working+Group

File: README.md (hunk 3 above)
Comment (iain-anderson) [reply README.md:101]: edgex-device-coap

File: res/configuration-native.toml (hunk 4 above)
Comment (iain-anderson) [reply res/configuration-native.toml:8]: 49988

File: VERSION (hunk 5 above)
Comment (iain-anderson) [reply VERSION:1]: policy doc here: https://wiki.edgexfoundry.org/pages/viewpage.action?pageId=21823969

Comment (iain-anderson) [reply VERSION:1]: Note that we're moving away from having VERSION files in the repository - instead the version information is generated at build time based on the most recent tag. Have a look at https://github.com/edgexfoundry/device-grove-c/blob/master/Makefile to see the sort of thing.

Comment (kb2ma) [reply VERSION:1]: I am working on an update to use the new versioning mechanism, and I'm confused about the git version tag. For example, commit edgexfoundry/device-grove-c@ee944e5eaf0fdadf041172ba91253cadf7663a84 is tagged as `v1.2.1`. Inclusion of the letter 'v' means the tag for the Docker image created by the Makefile `docker` target also includes the 'v' . For device-coap-c, that means the image tag will be `edgexfoundry/docker-device-coap-c:v0.2-dev`. Historically Docker image tags have not included the 'v'. So, should I just format the git version tag as device-grove-c does and be happy?

Comment (iain-anderson) [reply VERSION:1]: That's a good point. We should probably `sed` out the 'v' when generating VERSION in the Makefile.
(The official release mechanisms use other means to generate VERSION, so the published images are tagged without the 'v').
Other services default to a version of `0.0.0` in the Makefile, I was trying to do better :)

**Comment by kb2ma** [comment]
I just pushed some fixups, but I've got a better idea for the VERSION implementation. I'll force push later today. I'll let you know when ready for review.

**Comment by kb2ma** [comment]
Fixups ready for review, @iain-anderson. I'm happy to squash down to a single commit when you're done if that's customary.

File: VERSION (hunk 5 above)
Comment (kb2ma) [reply VERSION]: Sounds good; added pipe to sed to generate VERSION.

**Review by iain-anderson — APPROVED** [review]
Looks good. I think squash-merge for this will be fine

**Comment by kb2ma** [comment]
I have squashed manually in the past and was not aware of the squash-merge option. Thanks for review and the suggestion.

**Pull request merged**